docs/mp/jwt.adoc

```diff
@@ -55,6 +71,99 @@ public class ProtectedApplication extends Application{
 }
 ----
 
-Learn more about JWT authentication: +
+== API
+
+The following interfaces and annotations are used to work with JWT in Helidon MP:
+
+* `JsonWebToken` - an interface used in CDI beans _(@RequestScoped)_ dependency injection to obtain the JWT of the currently executing caller.
+* `@Claim` - an annotation used by CDI bean _(@RequestScoped)_  dependency injection to obtain individual claims from the caller’s JWT.
+* `ClaimValue` - a proxy interface used with `@Claim` annotation to оbtain the value of a claim by calling `getValue()`.
+
+== Configuration
+
+include::{rootdir}/config/io_helidon_microprofile_jwt.adoc[leveloffset=+1,tag=config]
+
+A configuration example in `microprofile-config.properties`:
+[source, properties]
+----
+mp.jwt.verify.issuer=https://{PublicIssuerDomain}/oauth2/default
+mp.jwt.verify.publickey.location=${mp.jwt.verify.issuer}/v1/keys
+----
+
+== Examples
+
+[source, java]
+----
+@Path("/hello")
+public class HelloResource {
+
+    @GET
+    @Produces(TEXT_PLAIN)
+    public String hello(@Context SecurityContext context) {
+        Optional<Principal> userPrincipal = context.userPrincipal();
+        return "Hello, " + userPrincipal.get().getName() + "!";
+    }
+}
+----
+
+Do not forget to annotate the `HelloApplication` class to enable JWT:
+
+[source, java]
+----
+@LoginConfig(authMethod = "MP-JWT")
+@ApplicationScoped
+public class HelloApplication extends Application {
+
+    @Override
+    public Set<Class<?>> getClasses() {
+        return Set.of(HelloResource.class);
+    }
+}
+----
+
+Add the following configuration in `microprofile-config.properties`:
+
+[source, properties]
+----
+mp.jwt.verify.issuer=https://{IssuerPublicDomain}/oauth2/default
+mp.jwt.verify.publickey.location=${mp.jwt.verify.issuer}/v1/keys
+----
+
+Obtain the Security Token from external issuer:
+
+[source, bash]
+----
+TOKEN=sdf4dDSWFcswdsffDSasEgv...
+----
+
+Run the application and execute an http request against it:
 
+[source, bash]
+----
+curl -X GET -I -H "Authorization: Bearer $TOKEN" http://localhost:8080/hello
+
+----
+
+The result should be:
+
+[source, bash]
+----
+HTTP/1.1 200 OK
+Date: 08.06.2022 10:33:47 EEST
+connection: keep-alive
+content-length: 28
+
+Hello, secure@helidon.io!
+----
+
+which means that the request successfully passed authentication.
+
+== Additional Information
+
+Learn more about JWT authentication at: +
 link:{microprofile-jwt-base-url}#_introduction[Eclipse MicroProfile Interoperable JWT RBAC]
+
+== Reference
+
+* https://github.com/eclipse/microprofile-jwt-auth[MicroProfile JWT Auth GitHub Repository]
+* https://download.eclipse.org/microprofile/microprofile-jwt-auth-{spec-version}/microprofile-jwt-auth-spec-{spec-version}.html[MicroProfile JWT Auth Spec]
```

Comment:
I would make it the first bullet because it has more importance.

Reply:
done